purchase_order_qty_by_product_category/models/purchase_order_qty_by_product_category.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class PurchaseOrderQtyByProductCategory(models.Model):
+    _name = "purchase.order.qty.by.product.category"
+    _description = "PO quantities split by product category"
+
+    purchase_order_id = fields.Many2one(
+        "purchase.order", required=True, ondelete="cascade"
+    )
+    purchase_line_ids = fields.One2many(
+        "purchase.order.line", "qty_by_product_category_id"
+    )
+    category_id = fields.Many2one("product.category", required=True, ondelete="cascade")
+    qty_ordered = fields.Float(
+        digits="Product Unit of Measure", compute="_compute_quantities", store=True
+    )
+    qty_received = fields.Float(
+        digits="Product Unit of Measure", compute="_compute_quantities", store=True
+    )
+    qty_to_receive = fields.Float(
+        digits="Product Unit of Measure", compute="_compute_quantities", store=True
+    )
+    qty_uom_id = fields.Many2one("uom.uom")
+
+    @api.constrains("purchase_order_id", "category_id", "qty_uom_id")
+    def _check_unicity(self):
+        for rec in self:
+            if (
+                self.search_count(
+                    [
+                        ("purchase_order_id", "=", rec.purchase_order_id.id),
+                        ("category_id", "=", rec.category_id.id),
+                        ("qty_uom_id", "=", rec.qty_uom_id.id),
+                    ]
+                )
+                > 1
+            ):
+                raise ValidationError(
+                    _(
+                        "Qty group for order '%(order)s', category '%(categ)s'"
+                        " and UoM '%(uom)s' already exists"
+                    )
+                    % {
+                        "order": rec.purchase_order_id.name,
+                        "categ": rec.category_id.name,
+                        "uom": rec.qty_uom_id.name or _("undefined"),
+                    }
+                )
+
+    @api.depends(
+        "purchase_line_ids.product_qty",
```

Comment:
I think we should add `qty_uom_id` to the depends as well.

Reply:
@SilvioC2C what do you think about this comment?

Reply:
@quannguyen1560 @HviorForgeFlow honestly, I disagree: the `qty_uom_id` is not supposed to be updated once the record is created, therefore it doesn't make sense to add it to the dependencies: since it's not gonna be updated, it will only make the dependencies "heavier".